src/core_plugins/kibana/public/dashboard/dashboard_state_manager.js

```diff
@@ -210,12 +187,29 @@ export class DashboardStateManager {
     if (!this._areStoreAndAppStatePanelsEqual()) {
       const panels = getPanels(store.getState());
       this.appState.panels = [];
+      this.panelIndexPatternMapping = {};
```

Comment:
Rather than do anything fancy with detecting added vs removed vs changed panels, we're just clearing and rebuilding below